forc-plugins/forc-client/src/util/aws.rs

```diff
@@ -0,0 +1,269 @@
+use async_trait::async_trait;
+use aws_config::{default_provider::credentials::DefaultCredentialsChain, Region, SdkConfig};
+use aws_sdk_kms::config::Credentials;
+use aws_sdk_kms::operation::get_public_key::GetPublicKeyOutput;
+use aws_sdk_kms::primitives::Blob;
+use aws_sdk_kms::types::{MessageType, SigningAlgorithmSpec};
+use aws_sdk_kms::{config::BehaviorVersion, Client};
+use fuel_crypto::Message;
+use fuels::prelude::*;
+use fuels::types::bech32::{Bech32Address, FUEL_BECH32_HRP};
+use fuels::types::coin_type_id::CoinTypeId;
+use fuels::types::input::Input;
+use fuels_accounts::provider::Provider;
+use fuels_accounts::{Account, ViewOnlyAccount};
+use fuels_core::traits::Signer;
+
+/// AWS configuration for the `AwsSigner` to be created.
+/// De-facto way of creating the configuration is to load it from env.
+#[derive(Debug, Clone)]
+pub struct AwsConfig {
+    sdk_config: SdkConfig,
+}
+
+impl AwsConfig {
+    /// Load configuration from environment variables.
+    /// For more details see: https://docs.rs/aws-config/latest/aws_config/
+    pub async fn from_env() -> Self {
+        let loader = aws_config::defaults(BehaviorVersion::latest())
+            .credentials_provider(DefaultCredentialsChain::builder().build().await);
+
+        let loader = match std::env::var("E2E_TEST_AWS_ENDPOINT") {
```

Comment:
Is this the right endpoint for production?

Reply:
This is placed here so that AwsConfig can be configured for tests in the future. It doesn't effect anything unless it is set. And by default it is not set in CI/locally. So this is here to make this thing a little bit more testable.